soci/index_builder.go

```diff
@@ -83,17 +86,23 @@ func WithBuildToolIdentifierOption(buildToolIdentifier string) IndexBuilderOptio
 	}
 }
 
-// BuildIndex builds SOCI index for an image
+func WithPlatformOption(platform ocispec.Platform) IndexBuilderOption {
+	return func(b *IndexBuilder) error {
+		b.platform = platform
```

Comment:
So thinking through how someone would use the builder - if they wanted to build an index for every platform the caller would have to create a new builder for each, e.g.

```
for _, platform := range platforms {
    builder := soci.NewIndexBuilder(store, WithPlatform(platform))
    index := builder.Build(ctx, desc)
}
```

This says to me that platform is a property of a particular call to `Build` and not of the `IndexBuilder` itself. It seems more natural to add a `BuildOption` concept and put the platform there.

e.g. usage like:
```
builder := soci.NewIndexBuilder(store)
for _, platform := range platforms {
    index := builder.Build(WithPlatform(platform))
}
```

Does that sound reasonable or is that almost going back to just having a `soci.BuildIndex` function?

Reply:
+1 to the 2nd implementation 
```
index := builder.Build(WithPlatform(platform))
```

Reply:
That seems reasonable. I can introduce back the `BuildOption`.

Reply:
cc @qingshenhuang for Build function signature change